Automated cherry pick of #4028: Fix race conditions in NetworkPolicyController Cherry pick of #4028 on release-1.7

xliuxu opened the pull request:
#4028: Fix race conditions in NetworkPolicyController

For details on the cherry pick process, see the [cherry pick requests](https://github.com/antrea-io/antrea/blob/main/docs/contributors/cherry-picks.md) page.

xliuxu commented:
/test-all

antoninbas commented:
@xliuxu is this not an automatic cherry-pick? I don't see the `kind/cherry-pick` label.

xliuxu commented:
> @xliuxu is this not an automatic cherry-pick? I don't see the `kind/cherry-pick` label.

Yes. This should be an automatic cherry-pick. I accidentally pushed to a wrong repo and created this PR manually to check the e2e results. I have added the label and changed the title accordingly.

xliuxu commented:
/test-all
